content/news/2022-11-05-bevy-0.9/index.md

```diff
@@ -0,0 +1,1045 @@
+
+## Multiple Directional Lights
+
+<div class="release-feature-authors">authors: @kurtkuehnert</div>
+
+Bevy now supports multiple directional lights (the new limit is 10 at once). Much like we did for point lights, we will likely make this unbounded on platforms that support storage buffers in the future, but this was a nice first step that maintains compatibility on all platforms.
+
+![multiple directional](multiple_directional.png)
+
+## WGPU Upgrade
+
+- [Update `wgpu` to 0.14.0, `naga` to `0.10.0`, `winit` to 0.27.4, `raw-window-handle` to 0.5.0, `ndk` to 0.7][6218]
+
+## Sprite Rects
+
+- [Sprite: allow using a sub-region (Rect) of the image][6014]
+
+## Plugin Settings
+
+<div class="release-feature-authors">authors: @cart, @mockersf</div>
+
+In past versions of Bevy, "immutable" Plugin settings were represented as normal ECS resources, which were read as part of plugin init. This presented a number of problems:
+
+1. If a user inserted the plugin settings resource after the plugin was initialized, it would be silently ignored (and use the defaults instead)
+2. Users could modify the plugin settings resource after the plugin had been initialized. This created a false sense of control over settings that could no longer be changed.
+
+These were especially problematic and confusing for the `WindowDescriptor` resource, but it was a general problem.
+
+To resolve this, in **Bevy 0.9** we moved plugin settings onto the plugins themselves, and created new apis for overriding the default settings:
+
+```rust
+app.add_plugins(DefaultPlugins
+  .set(AssetPlugin {
+    watch_for_changes: true,
+    ..default()
+  })
+  .set(WindowPlugin {
+    window: WindowDescriptor {
+      width: 400.0,
+      ..default()
+    },
+    ..default()
+  })
+)
+```
+
+This makes the connection between the settings and the plugin clear, and differentiates these "plugin init" settings from "runtime configurable" settings (which are still represented as ECS resources).
+
+## Plugins are now unique by default
+
+<div class="release-feature-authors">authors: @mockersf</div>
+
+Plugins are now unique by default. Attempting to add a unique plugin to an app more than once will result in an error. Plugins that are not intended to be unique can override the default `is_unique` method:
+
+```rust
+impl Plugin for MyPlugin {
+  fn build(&self, app: &mut App) {
+    app.add_system(some_system);
+  }
+
+  fn is_unique(&self) -> bool {
+    false
+  }
+}
+```
+
+## Task Pool Panic Handling
+
+<div class="release-feature-authors">authors: @james7132</div>
+
+- [TaskPool Panic Handling][6443]
+
+## Task Pool: Nested Spawns on Scope
+
+<div class="release-feature-authors">authors: @hymm</div>
+
+Bevy's Task Pools now support "nested spawns on scopes":
+
+```rust
+let results = task_pool.scope(|scope| {
+    scope.spawn(async move {
+        scope.spawn(async move { 1 });
+        2
+    });
+});
+
+assert_eq!(results, vec![2, 1]);
```

Comment:
```suggestion
assert!(results.contains(&0));
assert!(results.contains(&1));
```

I found out the hard way that the ordering of the results is actually non deterministic here. https://github.com/bevyengine/bevy/issues/6306

Reply:
nice catch!